src/ServiceInsight/App.xaml.cs

```diff
@@ -25,6 +26,9 @@ public App()
         protected override void OnStartup(StartupEventArgs e)
         {
             LogTo.Information("Starting the application...");
+                    
+            if (SingleInstance.AlreadyRunning())
```

Comment:
why not use a mutex?
```
    using var mutex = new Mutex(true, "ServiceInsight", out var createdNew);
        if (!createdNew)
        {
               App.Current.Shutdown();
        } 
```

Reply:
@SimonCropp we decided to go with a mix of Win32 and mutex. There is no native way to bring the window into foreground when you launch a new instance.